pkg/aws/rds/rds.go

```diff
@@ -73,15 +74,25 @@ func (r RDS) CreateDBCluster(ctx context.Context, storage int64, instanceClass,
 	return r.CreateDBClusterWithContext(ctx, dbi)
 }
 
+func (r RDS) CreateDBInstanceInClusterForTest(ctx context.Context, restoredClusterID, instanceID, instanceClass, dbEngine, dbSubnetGroup string, publicAccess bool) (*rds.CreateDBInstanceOutput, error) {
+	dbi := &rds.CreateDBInstanceInput{
+		DBClusterIdentifier:  &restoredClusterID,
+		DBInstanceClass:      &instanceClass,
+		DBInstanceIdentifier: &instanceID,
+		DBSubnetGroupName:    &dbSubnetGroup,
+		Engine:               &dbEngine,
+		PubliclyAccessible:   aws.Bool(publicAccess),
+	}
+	return r.CreateDBInstanceWithContext(ctx, dbi)
+}
+
 func (r RDS) CreateDBInstanceInCluster(ctx context.Context, restoredClusterID, instanceID, instanceClass, dbEngine, dbSubnetGroup string) (*rds.CreateDBInstanceOutput, error) {
-	pa := true
 	dbi := &rds.CreateDBInstanceInput{
 		DBClusterIdentifier:  &restoredClusterID,
 		DBInstanceClass:      &instanceClass,
 		DBInstanceIdentifier: &instanceID,
 		DBSubnetGroupName:    &dbSubnetGroup,
 		Engine:               &dbEngine,
-		PubliclyAccessible:   &pa,
```

Comment:
What will be the default value if not specified?

Reply:
```
	// If DBSubnetGroupName isn't specified, and PubliclyAccessible isn't specified,
	// the following applies:
	//
	//    * If the default VPC in the target Region doesn’t have an internet gateway
	//    attached to it, the DB cluster is private.
	//
	//    * If the default VPC in the target Region has an internet gateway attached
	//    to it, the DB cluster is public.
	//
	// If DBSubnetGroupName is specified, and PubliclyAccessible isn't specified,
	// the following applies:
	//
	//    * If the subnets are part of a VPC that doesn’t have an internet gateway
	//    attached to it, the DB cluster is private.
	//
	//    * If the subnets are part of a VPC that has an internet gateway attached
	//    to it, the DB cluster is public.
	//
	// Valid for: Multi-AZ DB clusters only
```
this is what the docs says, now we would be falling in the second case i.e we are providing DBsubnetGroupName, so it depends on the subnet group